shared/DownBit.F90

```diff
@@ -208,8 +223,38 @@ subroutine pFIO_DownBit1D ( x, xr, nbits_to_keep, undef, flops, rc )
      xr_ptr = c_loc(xr(1))
      call c_f_pointer(xr_ptr, xr_tmp,[size(x),1])
 
-     call pFIO_Downbit2d(x_tmp(:,:), xr_tmp(:,:), nbits_to_keep, undef=undef, flops=flops, rc=rc)
-
-   end subroutine pFIO_Downbit1D
-
-end module pFIO_DownbitMod
+     call Downbit2d(x_tmp(:,:), xr_tmp(:,:), nbits_to_keep, undef=undef, flops=flops, mpi_comm=mpi_comm, rc=rc)
+
+   end subroutine Downbit1D
+
+   subroutine compute_min_max(array,min_value,max_value,undef_value,mpi_comm,rc)
+      real, intent(in)  :: array(:,:)
+      real, intent(out) :: min_value
+      real, intent(out) :: max_value
+      real, intent(in ) :: undef_value
+      integer, intent(in ) :: mpi_comm
+      integer, optional, intent(out) :: rc
+
+      real :: local_min_value, local_max_value
+      logical, allocatable :: undef_mask(:,:)
+      integer :: status
+
+      allocate(undef_mask(size(array,1),size(array,2)))
+      undef_mask = .false.
+      where(array /= undef_value) undef_mask = .true.
+
+      local_min_value = minval(array,mask=undef_mask)
+      local_max_value = maxval(array,mask=undef_mask)
+      if (mpi_comm /= MPI_COMM_NULL) then
+         call MPI_AllReduce(local_min_value,min_value,1,MPI_FLOAT,MPI_MIN,mpi_comm,status)
+         _VERIFY(status)
+         call MPI_AllReduce(local_max_value,max_value,1,MPI_FLOAT,MPI_MAX,mpi_comm,status)
+         _VERIFY(status)
+      else
+         min_value = local_min_value
+         max_value = local_max_value
+      end if
```

Comment:
Are you certain that `MPI_AllReduce()` does not do the "right thing" with a null communicator?

Reply:
Well, I could not find a definite answer, and did see some comments on the internet that suggest NULL does not work in this context.  So closing this.